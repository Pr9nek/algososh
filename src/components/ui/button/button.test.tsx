import { Button } from "./button";
import { render, fireEvent, screen } from '@testing-library/react';
const testRenderer = require('react-test-renderer');

describe('Button component', () => {
    it('rendering a button with text', () => {
        const tree = testRenderer.create(<Button text='Кнопка'/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('rendering a button without text', () => {
        const tree = testRenderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('rendering a disabled button', () => {
        const tree = testRenderer.create(<Button disabled={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('rendering a button with a loading indication', () => {
        render(<Button />);
        const tree = testRenderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('the correctness of calling the callback when clicking on the button', () => {
        const callBackMockFn = jest.fn();
        render(<Button onClick={callBackMockFn} />)
        fireEvent.click(screen.getByRole('button'))
        expect(callBackMockFn).toHaveBeenCalled();
    });

})