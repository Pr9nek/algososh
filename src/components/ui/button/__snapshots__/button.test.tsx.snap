// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component rendering a button with a loading indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component rendering a button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Button component rendering a button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component rendering a disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
